api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -139,57 +146,63 @@ const (
-	ForwardModeAlwaysForwardOnly ForwardMode = "AlwaysForwardOnly"
+	// XFCCForwardModeSanitize removes the XFCC header from the request. This is the default mode.
+	XFCCForwardModeSanitize XFCCForwardMode = "Sanitize"
+
+	// XFCCForwardModeForwardOnly forwards the XFCC header in the request if the client connection is mTLS.
+	XFCCForwardModeForwardOnly XFCCForwardMode = "ForwardOnly"
+
+	// XFCCForwardModeAppendForward appends the client certificate information to the request’s XFCC header and forward it if the client connection is mTLS.
+	XFCCForwardModeAppendForward XFCCForwardMode = "AppendForward"
+
+	// XFCCForwardModeSanitizeSet resets the XFCC header with the client certificate information and forward it if the client connection is mTLS.
+	// The existing certificate information in the XFCC header is removed.
+	XFCCForwardModeSanitizeSet XFCCForwardMode = "SanitizeSet"
+
+	// XFCCForwardModeAlwaysForwardOnly always forwards the XFCC header in the request, regardless of whether the client connection is mTLS.
+	XFCCForwardModeAlwaysForwardOnly XFCCForwardMode = "AlwaysForwardOnly"
 )
 
-// Specifies the fields in the client certificate to be forwarded on the x-forwarded-client-cert (XFCC) HTTP header
-// By default, x-forwarded-client-cert (XFCC) will always include By and Hash data
-// +kubebuilder:validation:Enum=Subject;Cert;Chain;Dns;Uri
-type ClientCertData string
+// XFCCCertData specifies the fields in the client certificate to be forwarded in the XFCC header.
+// +kubebuilder:validation:Enum=Subject;Cert;Chain;DNS;URI
+type XFCCCertData string
 
 const (
-	// Whether to forward the subject of the client cert.
-	ClientCertDataSubject ClientCertData = "Subject"
-	// Whether to forward the entire client cert in URL encoded PEM format.
-	// This will appear in the XFCC header comma separated from other values with the value Cert=”PEM”.
-	ClientCertDataCert ClientCertData = "Cert"
-	// Whether to forward the entire client cert chain (including the leaf cert) in URL encoded PEM format.
-	// This will appear in the XFCC header comma separated from other values with the value Chain=”PEM”.
-	ClientCertDataChain ClientCertData = "Chain"
-	// Whether to forward the DNS type Subject Alternative Names of the client cert.
-	ClientCertDataDNS ClientCertData = "Dns"
-	// Whether to forward the URI type Subject Alternative Name of the client cert.
-	ClientCertDataURI ClientCertData = "Uri"
+	// XFCCCertDataSubject is the Subject field of the current client certificate.
+	XFCCCertDataSubject XFCCCertData = "Subject"
+	// XFCCCertDataCert is the entire client certificate in URL encoded PEM format.
+	XFCCCertDataCert XFCCCertData = "Cert"
+	// XFCCCertDataChain is the entire client certificate chain (including the leaf certificate) in URL encoded PEM format.
+	XFCCCertDataChain XFCCCertData = "Chain"
+	// XFCCCertDataDNS is the DNS type Subject Alternative Name field of the current client certificate.
+	XFCCCertDataDNS XFCCCertData = "DNS"
```

Comment:
how does k8s deal with 3 letter acronyms for PascalCase ?

Reply:
Gateway API and EG have many existing examples of `HTTP`, `URL`, `UDP`, `TCP`, `DNS` etc.

But I can't find "Dns" or "Uri". Envoy does use PascalCase for all acronyms.

Reply:
thanks, https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.ProtocolType is a good example that highlights this